apps/emqx_resource/src/emqx_resource_buffer_worker.erl

```diff
@@ -337,7 +325,8 @@ resume_from_blocked(Data) ->
                     {next_state, running, Data}
             end;
         {expired, Ref, Batch} ->
-            IsAcked = ack_inflight(InflightTID, Ref, Id, Index),
+            WorkerPid = self(),
+            IsAcked = ack_inflight(InflightTID, Ref, Id, Index, WorkerPid),
```

Comment:
m.b. an inline fun `ack_inflight_from_worker(InflightTID, Ref, Id, Index)`?

Reply:
I have another PR underway that will also touch this function, I'll try to do it in that PR.